granule.go

```diff
@@ -28,14 +28,22 @@ type Granule struct {
 // GranuleMetadata is the metadata for a granule.
 type GranuleMetadata struct {
 	// least is the row that exists within the Granule that is the least.
-	// This is used for quick insertion into the btree, without requiring an iterator
+	// This is used for quick insertion into the btree, without requiring an
+	// iterator.
 	least atomic.Pointer[dynparquet.DynamicRow]
 
-	// size is the raw commited, and uncommited size of the granule. It is used as a suggestion for potential compaction
+	// size is the raw committed, and uncommitted size of the granule. It is
+	// used as a suggestion for potential compaction.
 	size *atomic.Uint64
 
-	// pruned indicates if this Granule is longer found in the index
+	// pruned indicates if this Granule is longer found in the index.
 	pruned *atomic.Uint64
+
+	// compacting indicates if this Granule is currently being compacted.
+	// TODO(asubiotto): Is this necessary? We append a sentinel node to the
+	// parts list when compacting, but I'm unsure how to "read" that without
+	// appending a part. What's the intention?
```

Comment:
@thorfour knows this best

Reply:
This isn't necessary, granules have the (now poorly named) `pruned` variable that prevents multiple go routines from compacting the same granule

Reply:
Ah, gotcha! Yes I guess I was confused by the naming and somehow my brain read the CAS operation at the top of compaction to just be a load operation (with the assumption that `pruned` was set after pruning). Might rename the variable then.

Reply:
Yea it used to be... We should rename it